src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -177,4 +172,16 @@ private function validateKey($key)
 
         return $key;
     }
+
+    private function createGenerator(array $keys)
```

Comment:
It's possible to do inline function but I thought from code style approach it's not a good way

```php
/**
     * {@inheritdoc}
     */
    public function getItems(array $keys = array())
    {
        foreach ($keys as $key) {
            $this->validateKey($key);
        }

        return (function (array $keys) {
            $f = $this->createCacheItem;
            $now = time();

            foreach ($keys as $key) {
                $isHit = isset($this->expiries[$key]) && ($this->expiries[$key] >= $now || !$this->deleteItem($key));
    
                yield $key => $f($key, $isHit ? $this->values[$key] : null, $isHit);
            }
        })($keys);
    }
```